IOMC/ParticleGuns/interface/FlatProtonLogKsiLogTGun.h

```diff
@@ -0,0 +1,75 @@
+#ifndef IOMC_ParticleGuns_FlatProtonLogKsiLogTGun_h
+#define IOMC_ParticleGuns_FlatProtonLogKsiLogTGun_h
+
+#include <string>
+#include "HepPDT/defs.h"
+#include "HepPDT/TableBuilder.hh"
+#include "HepPDT/ParticleDataTable.hh"
+
+#include "HepMC/GenEvent.h"
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "CLHEP/Random/JamesRandom.h"
+#include "CLHEP/Random/RandFlat.h"
+
+#include <memory>
+#include "boost/shared_ptr.hpp"
+
+using namespace edm;
+namespace HepMC {
+	class GenEvent;
+}
+
+
+class FlatProtonLogKsiLogTGun : public edm::EDProducer
```

Comment:
Seems to be very different from typical gun implementation, see i.e. inheritance here:
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/IOMC/ParticleGuns/interface/BaseFlatGunProducer.h

Check also new CTPPS gun, submitted to official CMSSW repo:
https://github.com/cms-sw/cmssw/pull/15096/files